syscall.go

```diff
@@ -111,6 +114,62 @@ func unmountShareDir() error {
 	return os.RemoveAll(containerMountDest)
 }
 
+func waitForBlockDevice(deviceName string) (err error) {
+	devicePath := filepath.Join(devPath, deviceName)
+
+	if _, err := os.Stat(devicePath); err == nil {
+		return nil
+	}
+
+	u := goudev.Udev{}
+
+	// Create a monitor listening on a NetLink socket
+	monitor := u.NewMonitorFromNetlink("udev")
+
+	// Add filter to watch for just block devices
+	if err := monitor.FilterAddMatchSubsystemDevtype("block", "disk"); err != nil {
+		return err
+	}
+
+	// Create done signal channel for signalling epoll loop on the monitor socket.
+	done := make(chan struct{})
+
+	// Create channel to signal when desired udev event has been received.
+	doneListening := make(chan bool)
+
+	// Start monitor goroutine
+	ch, _ := monitor.DeviceChan(done)
+
+	go func() {
+		agentLog.Infof("Started listening for udev events for block device hotplug")
+
+		// Check if the device already exists
+		if _, err := os.Stat(devicePath); err == nil {
+			agentLog.Infof("Device %s already hotplugged, quit listening", deviceName)
+			close(done)
+			close(doneListening)
+		} else {
+			for d := range ch {
+				agentLog.Infof("Event:", d.Syspath(), d.Action())
+				if d.Action() == "add" && filepath.Base(d.Devpath()) == deviceName {
+					agentLog.Infof("Hotplug event received for device %s", deviceName)
+					close(doneListening)
+					break
+				}
+			}
+		}
+	}()
+
+	select {
+	case <-doneListening:
+		return
+	case <-time.After(1 * time.Second):
```

Comment:
You should remove `1 *` since you want 1 second.

I think we could wait up to 3 seconds, and in this case, the code should be:
```go
case <-time.After(time.Duration(3) * time.Second):
```
to avoid linter errors.